src/main/java/jewellery/inventory/service/SaleService.java

```diff
@@ -164,10 +180,117 @@ private void deleteSaleIfProductsIsEmpty(Sale sale) {
 
   private ProductReturnResponseDto validateSaleAfterReturnProduct(
       Sale sale, Product productToReturn) {
-    if (sale.getProducts().isEmpty()) {
+    if (sale.getProducts().isEmpty() && sale.getResources().isEmpty()) {
       return productService.getProductReturnResponseDto(null, productToReturn);
     }
     return productService.getProductReturnResponseDto(
         saleMapper.mapEntityToResponseDto(sale), productToReturn);
   }
+
+  private ResourceReturnResponseDto validateSaleAfterReturnResource(
+      Sale sale, PurchasedResourceInUser resourceToReturn) {
+    if (sale.getResources().isEmpty() && sale.getProducts().isEmpty()) {
+      return saleMapper.mapToResourceReturnResponseDto(resourceToReturn, null);
+    }
+    return saleMapper.mapToResourceReturnResponseDto(
+        resourceToReturn, saleMapper.mapEntityToResponseDto(sale));
+  }
+
+  private void setFieldsOfResourcesAfterSale(Sale sale) {
+    List<PurchasedResourceInUser> resources = sale.getResources();
+    resources.forEach(
+        resource -> {
+          resource.setPartOfSale(sale);
+          resource.setOwner(sale.getBuyer());
+          resource.setSalePrice(
+              resource.getQuantity().multiply(resource.getResource().getPricePerQuantity()));
+        });
```

Comment:
We are not setting discount here. Is it set somewhere else? I saw we are not setting it for the products as well.

Reply:
I set discount in the mapper